### PR TITLE
React-hooks

### PR DESCRIPTION
**Pull request by kostili-tec**
1. Task: https://github.com/rolling-scopes-school/tasks/tree/master/react/modules/module03
2. Screenshot:

4. Deploy: https://kostili-tec-react-rs.netlify.app/
5. Done 02.04.2023 / deadline 03.04.2023
6. Score: 15 / 15

  - [x] Eslint should not show any errors or warnings.
  - [x] All the functionality should work.
  - [x] Test coverage for the whole app should be not less than 60%.

### CODE REVIEW

File: src/components/UI/SearchInput/SearchInput.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React, { FC, useEffect, useRef, useState } from 'react';
+import classes from './input.module.scss';
+
+const SearchInput: FC = () => {
+  const [inputState, setInputState] = useState('');
```
Comment (Chichmarin17) [inline src/components/UI/SearchInput/SearchInput.tsx:5]: Don't name your state as you did with class components, better use here smth like `[searchField, setSearchField]`

File: src/components/pages/AboutPage.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React, { FC } from 'react';
+
+const AboutPage: FC = () => {
+  return (
+    <div>
+      <h3 style={{ marginTop: '2rem' }}>
```
Comment (Chichmarin17) [inline src/components/pages/AboutPage.tsx:6]: No need for 2 html tags here, better put all your mock text just right into `div`

File: src/components/pages/FormPage.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import React, { FC, useState } from 'react';
+import FormCardView from '../UI/FormComponents/FormCardView';
+import FormFileds from '../UI/FormComponents/FormFields';
+import { IFormCardData } from '../../interfaces';
+
+const FormPage: FC = () => {
+  const [cards, setCards] = useState<Array<IFormCardData> | []>([]);
+  const [message, setMessage] = useState(false);
```
Comment (Chichmarin17) [inline src/components/pages/FormPage.tsx:8]: For boolean values name your variables as you ask questions, e.g. `hasMessageCreated`, `shouldShowMessage` etc

File: src/components/pages/MainPage.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import Card from '../Card';
+import { getProducts } from '../../utils/api';
+import { IProductsArray } from '../../interfaces';
+import reactSVG from '../../assets/react.svg';
+
+interface IState {
+  products: IProductsArray;
+}
+const MainPage: FC = () => {
+  const [products, setProducts] = useState<IState['products']>([]);
```
Comment (Chichmarin17) [inline src/components/pages/MainPage.tsx:12]: same here, why do you use interface for typing your state and pick only one value in the place you use your interface? just write IProducts (omit array from its name)

File: src/tests/FormComponents.test.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    expect(await screen.findByText(/you must specify the name/i)).toBeInTheDocument();
+    expect(await screen.findByText(/you must select a category/i)).toBeInTheDocument();
+    expect(await screen.findByText(/you must specify the price/i)).toBeInTheDocument();
+    expect(await screen.findByText(/you must specify the production date/i)).toBeInTheDocument();
+    expect(await screen.findByText(/you have to choose a gift/i)).toBeInTheDocument();
+    expect(await screen.findByText(/you have to choose a condition/i)).toBeInTheDocument();
+    expect(await screen.findByText(/you have to choose an image/i)).toBeInTheDocument();
+    expect(await screen.findByText(/you must fill in the description/i)).toBeInTheDocument();
+  });
+  it('test', () => {
```
Comment (Chichmarin17) [inline src/tests/FormComponents.test.tsx:116]: what does the name `test` mean?

File: src/vite-env.d.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+/// <reference types="vite/client" />
```
Comment (Chichmarin17) [inline src/vite-env.d.ts:1]: do you need this .env?

Comment (kostili-tec) [reply src/vite-env.d.ts:1]: If I remove this file, I will get a lot of errors related to importing files(

**Pull request merged**